libtiff/libtiff_ctypes.py

```diff
@@ -786,59 +831,54 @@ def read_tiles(self, dtype=np.uint8):
         num_irows = self.GetField("ImageLength")
         if num_irows is None:
             num_irows = 1
-        num_idepth = self.GetField("ImageDepth")
-        if num_idepth is None:
-            num_idepth = 1
-
-        if num_idepth == 1 and num_irows == 1:
-            # 1D
-            full_image = np.zeros((num_icols,), dtype=dtype)
-        elif num_idepth == 1:
-            # 2D
-            full_image = np.zeros((num_irows, num_icols), dtype=dtype)
-        else:
-            # 3D
-            full_image = np.zeros((num_idepth, num_irows, num_icols),
-                                  dtype=dtype)
-
-        tmp_tile = np.zeros((num_trows, num_tcols), dtype=dtype)
-        tmp_tile = np.ascontiguousarray(tmp_tile)
-        for z in range(0, num_idepth):
-            for y in range(0, num_irows, num_trows):
-                for x in range(0, num_icols, num_tcols):
-                    r = self.ReadTile(tmp_tile.ctypes.data, x, y, z, 0)
+        # this number includes extra samples
+        samples_pp = self.GetField('SamplesPerPixel')
+        if samples_pp is None:  # default is 1
+            samples_pp = 1
+        planar_config = self.GetField('PlanarConfig')
+        if planar_config is None:  # default is contig
+            planar_config = PLANARCONFIG_CONTIG
+
```

Comment:
Oh, indeed... I don't know what's the difference at the file format level though. Anyway, in that case, you need to make a different code. Almost identical with samples_pp = depth, and planar_config =PLANARCONFIG_SEPARATE, but just z, instead of sample.